polkadot/node/core/pvf/src/worker_intf.rs

```diff
@@ -150,7 +150,42 @@ where
 	F: FnOnce(&Path) -> Fut,
 	Fut: futures::Future<Output = Result<T, SpawnErr>> + 'static,
 {
-	let socket_path = tmppath(&format!("pvf-host-{}", debug_id))
+	/// Returns a path under [`std::env::temp_dir`]. The path name will start with the given prefix.
+	///
+	/// There is only a certain number of retries. If exceeded this function will give up and return
+	/// an error.
+	pub async fn tmppath(prefix: &str) -> io::Result<PathBuf> {
```

Comment:
The root cause was this function returning a valid path but not creating the directory. It should only be used by `with_transient_socket_path`, which requires a valid path but nothing existing at the path. Elsewhere we should be using`tempfile` to correctly create temporary files or directories, preventing this kind of mistake in the future.